### PR TITLE
update AssemblyScript & remove some legacy + improvments

### PR DESCRIPTION
**Pull request by MaxGraey**

### CODE REVIEW

**Comment by hugo-dc** [comment]
Thanks a lot for submitting this PR!, please take a look at the formatting errors, it ban be fixed with `npm run format:fix`

File: assembly/evm.ts [hunk 1]
```diff
@@ -5,24 +5,24 @@ import { Opcodes } from './opcodes'
...
   // stack size is 100 elements
   // each stack element is 32 bytes
-  let stackSize = 100
-  let stackElementSize = 32
+  const stackSize = 100
+  const stackElementSize = 32
   let stack = new ArrayBuffer(stackElementSize * stackSize)
   let stackPtr = changetype<usize>(stack)
-  let stackElements = Array.create<Uint8Array>(100)
+  let stackElements = new Array<Uint8Array>(stackSize)
```
Comment (hugo-dc) [inline assembly/evm.ts]: This is interesting, I remember I tried using `new Array<...`, but for some reason it causes a runtime error `unreachable`.

```
wasm://wasm/cf9f5466:51



RuntimeError: unreachable
    at assembly/token/processBlock (wasm-function[50]:831)
    at assembly/token/main (wasm-function[51]:49)
```

do you have any idea why this is happening?

File: assembly/main.ts [hunk 2]
```diff
@@ -50,7 +50,7 @@ export enum NodeType {
 }
 
 // for Map<UintArray,Node> binaryen toText generates function names with commas, which wabt doesn't like.
-var Trie = new Map<usize, Node>()
+let Trie = new Map<usize, Node>()
```
Comment (hugo-dc) [inline assembly/main.ts]: There exists exactly the same line in `token.ts` which needs to be changed from `var` to `let`.

File: assembly/evm.ts (hunk 1 above)
Comment (MaxGraey) [reply assembly/evm.ts]: This new changes which allow `new Array<T>()` without nullable elements. Currently null-check happened during get indexed access and throw error if element is null but T isn't nullable. So `Array.create` now deprecated.

**Comment by MaxGraey** [comment]
I'm working in blind because unfortunately instillation process can't finish properly on my machine (some build conflicts scrypt with latest node.js)

File: assembly/evm.ts (hunk 1 above)
Comment (hugo-dc) [reply assembly/evm.ts]: Thanks for the explanation, after reinstalling the dependencies again I don't get the runtime error anymore.

**Comment by hugo-dc** [comment]
@MaxGraey it looks great!, thanks again for the PR.

**Pull request merged**